rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java

```diff
@@ -2004,6 +2012,54 @@ void should_SetTransferFlagToTrue_When_TransferringWithoutRequestBody() {
       assertThat(response.getBody().getWorkbasketSummary().getWorkbasketId())
           .isEqualTo("WBI:100000000000000000000000000000000006");
       assertThat(response.getBody().isTransferred()).isTrue();
+      assertThat(response.getBody().getOwner()).isNull();
+    }
+
+    @TestFactory
+    Stream<DynamicTest>
+        should_SetTransferFlagAndOwnerDependentOnRequestBody_When_TransferringTasks() {
+
+      Iterator<Pair<Boolean, String>> iterator =
+          Arrays.asList(Pair.of(true, "user-1-1"), Pair.of(false, "user-1-2")).iterator();
+      String url =
+          restHelper.toUrl(
+              RestEndpoints.URL_TRANSFER_WORKBASKET_ID, "WBI:100000000000000000000000000000000006");
+
+      List<String> taskIds =
+          Arrays.asList(
+              "TKI:000000000000000000000000000000000003",
+              "TKI:000000000000000000000000000000000004",
+              "TKI:000000000000000000000000000000000039");
+
+      ThrowingConsumer<Pair<Boolean, String>> test =
+          pair -> {
+            HttpEntity<Object> auth =
+                new HttpEntity<>(
+                    new TransferTaskRepresentationModel(pair.getLeft(), pair.getRight(), taskIds),
+                    RestHelper.generateHeadersForUser("admin"));
+            ResponseEntity<Map<String, Object>> response =
+                TEMPLATE.exchange(url, HttpMethod.POST, auth, BULK_RESULT_TASKS_MODEL_TYPE);
+
+            assertThat(response.getBody()).isNotNull();
+            Map<String, LinkedHashMap> failedTasks =
+                (Map<String, LinkedHashMap>) response.getBody().get("tasksWithErrors");
```

Comment:
Why do you cast here instead of using the representation model?

Reply:
From my understanding, I think you want me to do this:

BulkOperationResultsRepresentationModel failedTasks = response.getBody();

I think we have discussed this before. We cannot do that. BulkOperationResultRepresentationModel uses Map<String, ErrorCode> to capture the tasks failed to be transferred. Unfortunately ErrorCode is not deserializable, which mean the json format of the response for the errorCode cannot be converted back to the ErrorCode object. This is why we cannot expect the response Body to be an instance of BulkOperationsResultsRepresentationModel